app/controllers/api/v1/sessions_controller.rb

```diff
@@ -15,11 +15,14 @@ def create
         user = User.find_by(email: session_params[:email])
 
         # TODO: Add proper error logging for non-verified token hcaptcha
-        return render_error if hcaptcha_enabled? && !verify_hcaptcha(response: params[:token])
+        # return render_error if hcaptcha_enabled? && !verify_hcaptcha(response: params[:token])
 
         # TODO: Add proper error logging for non-verified token hcaptcha
         if user.present? && user.authenticate(session_params[:password])
-          sign_in user
+          # Creates an extra cookie if extend_session is selected by the user.
+          cookies.encrypted[:_extended_session] = { value: user.id, expires: 14.days } if session_params[:extend_session]
```

Comment:
For security purposes we'd like to mark the `_extended_session` as `httponly` to prevent scripts from accessing it.
Also, I think you could leverage that `sign_in user` to do all of this and further clean the action.

Reply:
```
      # Signs In the user
      def sign_in(user:, extended:)
        # Creates an extra cookie if extend_session is selected by the user.
        if extended
          cookies.encrypted[:_extended_session] = {
            value: user.id,
            expires: 14.days,
            httponly: true
          }
        end

        session[:user_id] = user.id
      end
```